Core/LocalizationManager.cs

```diff
@@ -178,58 +225,70 @@ public IEnumerable<CultureInfo> GetCultures()
 			return lst;
 		}
 
-		public void Delete(params ILocalizedObject[] list)
+		public void Delete(CultureInfo culture, params ILocalizedObject[] list)
 		{
 			Repository.Delete(list);
-			LocalizationCache.Clear();
+			RemoveFromCache(culture, list);
+		}
+
+		private void RemoveFromCache(CultureInfo culture, ILocalizedObject[] list)

```

Comment:
There is already method called 'Delete', I guess just 'Remove' will be not enough. This class is responsible for all localization processing, so it's better to specify that we're clearing cache.